app/screens/theme/theme.js

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View, FlatList} from 'react-native';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+
+import StatusBar from 'app/components/status_bar';
+import Section from 'app/screens/settings/section';
+import SectionItem from 'app/screens/settings/section_item';
+import FormattedText from 'app/components/formatted_text';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import Preferences from 'mattermost-redux/constants/preferences';
+
+export default class Theme extends React.PureComponent {
+    static propTypes = {
+        teamId: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        userId: PropTypes.string.isRequired,
+        actions: PropTypes.shape({
+            savePreferences: PropTypes.func.isRequired,
+        }).isRequired,
+        allowedThemes: PropTypes.arrayOf(PropTypes.object),
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    setTheme = (key) => {
+        const {userId, teamId, actions: {savePreferences}, allowedThemes} = this.props;
+        const selectedTheme = allowedThemes.find((theme) => theme.key === key);
+
+        savePreferences(userId, [{
+            user_id: userId,
+            category: Preferences.CATEGORY_THEME,
+            name: teamId,
+            value: JSON.stringify(selectedTheme),
+        }]);
+    }
+
+    renderThemeRow = ({item}) => {
+        const {theme} = this.props;
+        const style = getStyleSheet(theme);
+
+        return (
+            <View>
```

Comment:
We don't need to wrap these in a `<View>` any more when you're just using it to return multiple components. If you use `<React.Fragment>`, it does the same thing without rendering anything for the wrapper. You might also be able to return an array of components in the newest version of RN, but I haven't tested that yet.